### DISCUSSION TITLE
Are there known downsides/incompatibilities by using the current commit with python 3.9?

### DISCUSSION THREAD

**Discussion by ChrisOKay** [PicoQuant/snAPI#2] [Q&A] [answered]
When using version the alpha version 0.3, the [union type x | y](https://peps.python.org/pep-0604/) pipe operate was used in the source code, which made it necessary to use at least python 3.10 (or modify the source code accordingly).
This seems to have been replaced by the "old fashioned" Union[x, y] [two weeks ago](https://github.com/PicoQuant/snAPI/commit/544a9d18c42a4cd786982314c3987acd397fc1ad) (not part of any release yet), which is good for us, because we are still using python 3.9.

Since the installation guide still [advises to install python 3.11](https://picoquant.github.io/snAPI/install.html): are there known downsides/incompatibilities by using the current snapi commit with python 3.9?

**Comment by tpoint75** [comment, marked as answer]
I don't test it with older version. But lately I used snAPI with spyder which is using a Python 3.8.10 kernel. Therefore I had to make a fix that is not in a released but it should work for you:
https://github.com/PicoQuant/snAPI/commit/544a9d18c42a4cd786982314c3987acd397fc1ad

**Reply by ChrisOKay** [reply]
That sounds promising! I would appreciate to integrate this fix permanently in future releases to keep a "low entrance hurdle" when using snapi in established production environments (where a change in python version may cause headaches).

**Reply by tpoint75** [reply]
It will be in the next release! Thank you for helping us making snAPI better!
